test/any/web_catalog/api-extractor-test.es6.js

```diff
@@ -0,0 +1,331 @@
+          'property': 3,
+          'object': 7,
+          'buildInProperty': 3,
+        },
+      },
+      'metadata': {
+        '10000': {
+          'constantNumber': {
+            writable: 0,
+          },
+          'constantObject': {
+            writable: 0,
+          },
+          'property': {
+            writable: 1,
+          },
+          'object': {
+            writable: 1,
+          },
+          'buildInProperty': {
+            writable: 1,
+          },
+        },
+      },
+      types,
+    });
+    let properties = extractor.getObjectProperties_(og, 10000);
+    extractor.buildInObjectProperties = ['buildInProperty'];
+    it('captures non-constant properties', function() {
+      expect(properties).toContain('property');
+    });
+    it('captures object regardless if it is constant or not', function() {
+      expect(properties).toContain('object');
+    });
+    it('excludes constant property', function() {
+      expect(properties).not.toContain('constantNumber');
+    });
+    it('excludes non-own-property', function() {
+      expect(properties).not.toContain('nonOwnProperty');
+    });
+    it('does not filter build-in properties', function() {
+      expect(properties).toContain('buildInProperty');
+    });
+    // Turn on retainConstantMembers flag.
+    extractor.defaults.retainConstantMembers = true;
+    let propertiesWithConst = extractor.getObjectProperties_(og, 10000);
+    it('contains constant property if retainConstantMembers is true',
+    function() {
+      expect(propertiesWithConst).toContain('constantNumber');
+    });
+  });
+
+  describe('Method cleanUp', function() {
+    it('remves all + and $ characters', function() {
+      expect(extractor.cleanUp_('+toString+')).toEqual('toString');
+      expect(extractor.cleanUp_('$prototype$')).toEqual('prototype');
+    });
+  });
+
+  describe('Method isPosotiveInt', function() {
+    it('correctly identify a number string to be true', function() {
+      expect(extractor.isPosotiveInt_('0')).toEqual(true);
+      expect(extractor.isPosotiveInt_('1')).toEqual(true);
+      expect(extractor.isPosotiveInt_('2')).toEqual(true);
+      expect(extractor.isPosotiveInt_('3')).toEqual(true);
+      expect(extractor.isPosotiveInt_('10')).toEqual(true);
+      expect(extractor.isPosotiveInt_('999')).toEqual(true);
+    });
+    it('will not identify a non-number string to be false', function() {
+      expect(extractor.isPosotiveInt_('$1')).toEqual(false);
+      expect(extractor.isPosotiveInt_('toString')).toEqual(false);
+    });
+  });
+
+  describe('Method arrayMerge', function() {
+    it('merges arrays into first array with no duplicates', function() {
+      let array1 = ['a', 'b'];
+      let array2 = ['a', 'c', 'd'];
+      let array3 = ['b', 'd', 'f'];
+      extractor.arrayMerge_(array1, array2, array3);
+      expect(array1).toEqual(['a', 'b', 'c', 'd', 'f']);
+    });
+    it('filter out number(index) when merge', function() {
```

Comment:
nit: `filters`.

Also, do we document _why_ what appears to be an abstract merging function special-cases numbers? It's fine that it does, but we want to make sure it's well documented.